### ISSUE TITLE
Unable to Query Partitioned COW HUDI tables using Hudi Connector

### ISSUE THREAD

**Issue by Wrekkers** [trinodb/trino#15368]
Our team is testing the Hudi Connector on COPY ON WRITE  HUDI tables. And we are facing the error while reading from partitioned tables. 

`Caused by: java.lang.NoClassDefFoundError: Could not initialize class org.apache.hudi.common.bootstrap.index.HFileBootstrapIndex`

Trino version: v400
HUDI version: 0.10.1

**Trino Hudi Connector Properties:**
```
connector.name=hudi
hive.metastore.uri={METASTORE_URI}
hive.s3.iam-role={S3_IAM_ROLE}
hive.metastore-refresh-interval=2m
hive.metastore-timeout=3m
hudi.max-outstanding-splits=1800
hive.s3.max-error-retries=50
hive.s3.con
nect-timeout=1m
hive.s3.socket-timeout=2m
hudi.parquet.use-column-names=true
hudi.metadata-enabled=true
```

**Hudi Properties set while writing:**
```
hoodie.datasource.write.partitionpath.field = "insert_ds_ist",
hoodie.datasource.write.recordkey.field = "id",
hoodie.datasource.write.precombine.field = "_hoodie_incremental_key", (self generated column),
hoodie.datasource.write.hive_style_partitioning = "true",
hoodie.datasource.hive_sync.auto_create_database = "true",
hoodie.parquet.compression.codec = "gzip",
hoodie.table.name = "<table_name>",
hoodie.datasource.write.keygenerator.class = "org.apache.hudi.keygen.SimpleKeyGenerator",
hoodie.datasource.write.table.type = "COPY_ON_WRITE",
hoodie.metadata.enable = "true",
hoodie.datasource.hive_sync.enable = "true",
hoodie.datasource.hive_sync.partition_fields = "insert_ds_ist",
hoodie.datasource.hive_sync.partition_extractor_class = "org.apache.hudi.hive.MultiPartKeysValueExtractor"
```

General information of table:
Total rows = 1,213,959,199
Total Partitions = 2400+
Total file objects = 120,000
Total Size on S3 = 12~13 GB

Coordinator Relevant Logs:
[Partitioned_COW_Hudi_Coordinator_logs.log](https://github.com/trinodb/trino/files/10206457/Partitioned_COW_Hudi_Coordinator_logs.log)

**Comment by shubhamg931** [comment]
@codope @vinothchandar can we check this? This seems like a very basic test case of Trino Hudi Connector

**Comment by codope** [comment]
There could be a class conflict. We do have a product test for this simple scenario. 
Can you give the following suggestions a try:
1. Disable metadata and query again.
2. With metadata enabled, provide [hudi-trino-bundle](https://mvnrepository.com/artifact/org.apache.hudi/hudi-trino-bundle) to the plugin i.e. inside <trino_install_dir>/plugin/hudi.

**Comment by shubhamg931** [comment]
Hey, we did check both cases, it is working in the 1st case, but the query is too slow which is why we wanted to enable metadata in the first place

We did try 2nd suggestion as well, still getting the same error

FYI we are using spark2.4 to write the hudi tables, can it be related to that?

**Comment by shubhamg931** [comment]
@codope have checked this with spark 3.3.0 as well, still not working. Can you please let me know the configuration of the testing environment where this was tested?

**Comment by jianghuazhu** [comment]
Yes, we have encountered the same problem, and hope to join in the discussion.
